### PR TITLE
(no slip) NSAlert modal on bad input string

### PR DESCRIPTION
**Pull request by theoretick**
Present error modal on invalid input instead of crashing..

### CODE REVIEW

File: MapPointMapper/Parser/Parser.swift [hunk 1]
```diff
@@ -44,10 +48,15 @@ class Parser {
 
   - returns: An array of `CLLocationCoordinate2D` arrays representing the parsed areas/lines
   */
-  class func parseString(input: NSString, longitudeFirst: Bool) -> [[CLLocationCoordinate2D]] {
-    return Parser(longitudeFirst: longitudeFirst).parseInput(input)
+  class func parseString(input: NSString, longitudeFirst: Bool) throws -> [[CLLocationCoordinate2D]] {
+    let coordinate_set = Parser(longitudeFirst: longitudeFirst).parseInput(input)
```
Comment (dmiedema) [inline MapPointMapper/Parser/Parser.swift:52]: could `guard` be used here?

File: MapPointMapper/ViewController.swift [hunk 2]
```diff
@@ -203,9 +211,20 @@ class ViewController: NSViewController, MKMapViewDelegate, NSTextFieldDelegate {
 
   - parameter input: `NSString` to parse and draw on the map. If no string is given this is essentially a noop
```
Comment (dmiedema) [inline MapPointMapper/ViewController.swift:212]: probably wanna update docs with return value now (its private so importance is meh)

File: MapPointMapper/Parser/Parser.swift [hunk 3]
```diff
@@ -212,7 +221,11 @@ class Parser {
     let loc = range?.location as Int!
     let len = range?.length as Int!
     
-    return input.substringWithRange(NSRange(location: loc, length: len)) as NSString
+    if (loc != nil && len != nil) {
```
Comment (dmiedema) [inline MapPointMapper/Parser/Parser.swift]: i feel like `guard` could be used here too ¯\_(ツ)_/¯

File: MapPointMapper/Parser/Parser.swift (hunk 1 above)
Comment (theoretick) [inline MapPointMapper/Parser/Parser.swift:52]: :+1:

File: MapPointMapper/ViewController.swift (hunk 2 above)
Comment (theoretick) [inline MapPointMapper/ViewController.swift:212]: good call. done

File: MapPointMapperTests/Parser/ParserTests.swift [hunk 4]
```diff
@@ -70,6 +70,15 @@ class ParserSpec: XCTestCase {
     XCTAssertEqual(first.first!, second.first!, "First item should equal second, first was \(first.first!), second was \(second.first!)")
     XCTAssertEqual(first.last!, second.last!, "First item should equal second, first was \(first.last!), second was \(second.last!)")
   }
+  
+  func testParsingInvalidInputString() {
+    let inputString = "POLYGON (( herp derp dee derp"
+    do {
+      try Parser.parseString(inputString, longitudeFirst: false)
+    } catch {
+      XCTAssertTrue(error is ErrorType)
```
Comment (theoretick) [inline MapPointMapperTests/Parser/ParserTests.swift:79]: I want `XCTAssertThrows` but it dont exist apparently :-/  See [Potential Improvement #2](http://modocache.io/xctest-the-good-parts)

**Pull request merged**